packages/glimmer-syntax/lib/traversal/walker.ts

```diff
@@ -1,50 +1,53 @@
-function Walker(order=undefined) {
-  this.order = order;
-  this.stack = [];
-}
-
-export default Walker;
-
-Walker.prototype.visit = function(node, callback) {
-  if (!node) {
-    return;
-  }
-
-  this.stack.push(node);
-
-  if (this.order === 'post') {
-    this.children(node, callback);
-    callback(node, this);
-  } else {
-    callback(node, this);
-    this.children(node, callback);
-  }
-
-  this.stack.pop();
-};
-
 let visitors = {
-  Program: function(walker, node, callback) {
+  Program(walker, node, callback) {
```

Comment:
If you can type these it would be great, but it's optional (I can see that it might be very difficult)